insalan/tournament/models/seat_slot.py

```diff
@@ -0,0 +1,38 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+
+class SeatSlot(models.Model):
+    """
+    Represents a single seat
+    """
+
+    tournament = models.ForeignKey(
+        "Tournament",
+        null=False,
+        blank=False,
+        on_delete=models.CASCADE,
+        verbose_name=_("Tournoi"),
+    )
+
+    seats = models.ManyToManyField(
+        "Seat",
+        verbose_name=_("Place"),
+    )
+
+    class Meta:
+        """Meta Options"""
+
+        verbose_name = _("Slot")
+        verbose_name_plural = _("Slots")
+
+    def clean(self):
```

Comment:
When calling the clean method, django still doesn't have created the object.
ManyToMany fields are maps in the DB represented as ID -> ID.
So, when you got into this function, the ID doesn't exist and so isn't the ManyToMany field.
It results in an error "<SeatSlot: SeatSlot object (None)>" needs to have a value for field "id" before this many-to-many relationship can be used."

Reply:
Good catch. It seems the proper way to validate this is with a `ModelForm`.